solidity/contracts/converter/BancorConverterRegistry.sol

```diff
@@ -76,14 +78,113 @@ contract BancorConverterRegistry is IBancorConverterRegistry, ContractRegistryCl
     }
 
     /**
-      * @dev adds a converter to the registry
-      * anyone can add a converter to the registry, as long as the converter is active and valid
+      * @dev creates a liquid-token converter and adds it to the registry
+      * 
+      * @param _smartTokenName token name
+      * @param _smartTokenSymbol token symbol
+      * @param _smartTokenDecimals token decimals
+      * @param _maxConversionFee maximum conversion-fee
+      * @param _reserveToken reserve token
+      * @param _reserveRatio reserve ratio
+      * @param _reserveAmount reserve amount
+    */
+    function newLiquidToken(
+        string _smartTokenName,
+        string _smartTokenSymbol,
+        uint8 _smartTokenDecimals,
+        uint32 _maxConversionFee,
+        IERC20Token _reserveToken,
+        uint32 _reserveRatio,
+        uint256 _reserveAmount
+    )
+    public payable
+    {
+        IBancorConverterFactory factory = IBancorConverterFactory(addressOf(BANCOR_CONVERTER_FACTORY));

```

Comment:
need to move all that logic to a different contract so that we can upgrade it without upgrading the registry

Reply:
@yudilevi : The converter registry? I thought that was the whole idea about splitting it into a data contract and a logic contract (this contract) - in order to avoid having to update the data contract and migrate all of the data in it. Now we want to also avoid updating the logic contract?

Reply:
discussed already, we'll open a separate task for this

Reply:
@yudilevi: I already have.

Reply:
kk